ui/index.d.ts

```diff
@@ -1017,39 +1018,30 @@ export interface UIExtensionVariables {
     reporter: ITelemetryReporter;
 }
 
-export interface IAddUserAgent {
-    addUserAgentInfo(additionalUserAgentInfo: any): void;
-}
-
 /**
  * Retrieves a user agent string specific to the VS Code extension, of the form `${extensionName}/${extensionVersion}`,
  * and appends it to the given user agent string, if it isn't already in the string. Passing in no existingUserAgent
  * will return just the extension portion to use in a user agent.
  */
 export declare function appendExtensionUserAgent(existingUserAgent?: string): string;
 
-/**
- * Adds the extension user agent to the given ServiceClient or other object support AddUserAgentInfo
- */
-export declare function addExtensionUserAgent(client: IAddUserAgent): void;
-
 /**
  * Creates an Azure client, ensuring best practices are followed. For example:
  * 1. Adds extension-specific user agent
  * 2. Uses resourceManagerEndpointUrl to support sovereigns
  */
-export function createAzureClient<T extends IAddUserAgent>(
-    clientInfo: { credentials: ServiceClientCredentials; subscriptionId: string; environment: AzureEnvironment; },
-    clientType: new (credentials: ServiceClientCredentials, subscriptionId: string, baseUri?: string, options?: AzureServiceClientOptions) => T): T;
+export function createAzureClient<T>(
+    clientInfo: { credentials: ServiceClientCredentials; subscriptionId: string; environment: EnvironmentParameters; },
```

Comment:
What's the difference between `EnvironmentParameters` and `Environment`? Why doesn't this match the type on `ISubscriptionContext`?

Reply:
`EnvironmentParameters` is an interface for all the fields required to invoke the constructor for the `Environment` class. 

The `EnvironmentParameters` interface has just readonly primitive fields - one boolean, everything else string. 
The `Environment` class has methods like add and get, and a few more fields. 

I've aimed to use just the parameters wherever just the type is required, and Environment otherwise.

Reply:
If you want to go even further down that path, you could just use this as the type:
```
    clientInfo: { credentials: ServiceClientCredentials; subscriptionId: string; environment: { resourceManagerEndpointUrl: string }; }
```